source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -0,0 +1,375 @@
+      Http::Headers::get().AccessControlAllowOriginValue.All);
+  response_headers.insertNoChunks().value().setReference("0");
+
+  Http::StreamDecoderFilterCallbacks* stream_decoder_filter_callbacks =
+      admin_filter.getDecoderFilterCallbacks();
+
+  registerConnection(stream_decoder_filter_callbacks);
+
+  // Separated out just so it's easier to understand
+  auto on_destroy_callback = [this, stream_decoder_filter_callbacks]() {
+    // Unregister the callbacks from the sink so data is no longer encoded through them.
+    unregisterConnection(stream_decoder_filter_callbacks);
+  };
+
+  // Add the callback to the admin_filter list of callbacks
+  admin_filter.addOnDestroyCallback(std::move(on_destroy_callback));
+
+  ENVOY_LOG(debug, "start sending data to hystrix dashboard on port {}",
+            stream_decoder_filter_callbacks->connection()->localAddress()->asString());
+  return Http::Code::OK;
+}
+
+void HystrixSink::beginFlush() { current_stat_values_.clear(); }
+
+void HystrixSink::flushCounter(const Stats::Counter& counter, uint64_t) {
+  if (callbacks_list_.empty()) {
+    return;
+  }
+  if (counter.name().find("upstream_rq_") != std::string::npos) {
+    current_stat_values_[counter.name()] = counter.value();
+  }
+}
+// void HystrixSink::flushGauge(const Gauge& gauge, uint64_t value);
+void HystrixSink::endFlush() {
+  if (callbacks_list_.empty())
+    return;
+  stats_->incCounter();
+  for (auto& cluster : server_->clusterManager().clusters()) {
+    stats_->updateRollingWindowMap(current_stat_values_, cluster.second.get().info()->name());
+  }
+  std::stringstream ss;
+  for (auto& cluster : server_->clusterManager().clusters()) {
+    stats_->getClusterStats(
+        ss, cluster.second.get().info()->name(),
+        cluster.second.get()
+            .info()
+            ->resourceManager(Upstream::ResourcePriority::Default)
+            .pendingRequests()
+            .max(),
+        server_->stats()
+            .gauge("cluster." + cluster.second.get().info()->name() + ".membership_total")
+            .value(),
+        server_->statsFlushInterval().count());
+  }
+  Buffer::OwnedImpl data;
+  for (auto callbacks : callbacks_list_) {
+    data.add(ss.str());
+    callbacks->encodeData(data, false);
+  }
+
+  // send keep alive ping
+  // TODO (@trabetti) : is it ok to send together with data?
+  Buffer::OwnedImpl ping_data;
+  for (auto callbacks : callbacks_list_) {
+    ping_data.add(":\n\n");
+    callbacks->encodeData(ping_data, false);
+  }
+}
+
+void HystrixSink::registerConnection(Http::StreamDecoderFilterCallbacks* callbacks_to_register) {
+  callbacks_list_.emplace_back(callbacks_to_register);
+}
+
+void HystrixSink::unregisterConnection(Http::StreamDecoderFilterCallbacks* callbacks_to_remove) {
+  for (auto it = callbacks_list_.begin(); it != callbacks_list_.end();) {
+    if ((*it)->streamId() == callbacks_to_remove->streamId()) {
+      it = callbacks_list_.erase(it);
```

Comment:
Why not `break` after erasing?  Is it possible to have multiple callbacks with the same stream id here?  If it is, are you sure you want to delete both?  Maybe it makes sense to de-dup in the `registerConnection()` method so this can't happen?  What do you think?